### PR TITLE
table: fix err check

### PR DESCRIPTION
**Pull request by Tjianke**
### What problem does this PR solve?

Issue Number: fix #22979  

Problem Summary:

### What is changed and how it works?

What's Changed: added err check

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

- No release note

### CODE REVIEW

File: table/column_test.go [hunk 1]
```diff
@@ -104,12 +104,18 @@ func (t *testTableSuite) TestCheck(c *C) {
...
+	c.Assert(err, NotNil)
 	cols = cols[:1]
-	CheckNotNull(cols, types.MakeDatums(nil))
+	err = CheckNotNull(cols, types.MakeDatums(nil))
+	c.Assert(err, IsNil)
 	cols[0].Flag |= mysql.NotNullFlag
-	CheckNotNull(cols, types.MakeDatums(nil))
-	CheckOnce([]*Column{})
+	err = CheckNotNull(cols, types.MakeDatums(nil))
+	// to fix:
```
Comment (zimulala) [inline table/column_test.go:115]: Could we remove this comment?

File: table/column_test.go [hunk 2]
```diff
@@ -104,12 +104,18 @@ func (t *testTableSuite) TestCheck(c *C) {
 	col := newCol("a")
 	col.Flag = mysql.AutoIncrementFlag
 	cols := []*Column{col, col}
-	CheckOnce(cols)
+	// to fix:
```
Comment (zimulala) [inline table/column_test.go:107]: Ditto.

Comment (Tjianke) [reply table/column_test.go]: Removed.

File: table/column_test.go [hunk 3]
```diff
@@ -104,12 +104,16 @@ func (t *testTableSuite) TestCheck(c *C) {
 	col := newCol("a")
 	col.Flag = mysql.AutoIncrementFlag
 	cols := []*Column{col, col}
-	CheckOnce(cols)
+	err := CheckOnce(cols)
+	c.Assert(err, NotNil)
```
Comment (zz-jason) [inline table/column_test.go:108]: please check whether the error message is expected, not only check whether is an error message.

Comment (Tjianke) [reply table/column_test.go:108]: Added err msg check.

**Pull request merged**